packages/core/src/Actor.ts

```diff
@@ -49,57 +56,327 @@ export function createNullActor(id: string): ActorRef<any> {
   };
 }
 
-/**
- * Creates a deferred actor that is able to be invoked given the provided
- * invocation information in its `.meta` value.
- *
- * @param invokeDefinition The meta information needed to invoke the actor.
- */
-export function createInvocableActor<TC, TE extends EventObject>(
-  invokeDefinition: InvokeDefinition<TC, TE>,
-  machine: StateMachine<TC, any, TE, any>,
-  context: TC,
-  _event: SCXML.Event<TE>
-): ActorRef<any> {
-  const invokeSrc = toInvokeSource(invokeDefinition.src);
-  const serviceCreator = machine?.options.services?.[invokeSrc.type];
-  const resolvedData = invokeDefinition.data
-    ? mapContext(invokeDefinition.data, context, _event)
-    : undefined;
-  const tempActor = serviceCreator
-    ? createDeferredActor(
-        serviceCreator as Spawnable,
-        invokeDefinition.id,
-        resolvedData
-      )
-    : createNullActor(invokeDefinition.id);
-
-  // @ts-ignore
-  tempActor.meta = invokeDefinition;
-
-  return tempActor;
+interface SpawnOptions {
+  name: string;
+  autoForward: boolean;
+  sync: boolean;
 }
 
+export const wrappedMachines = new WeakSet();
+
+type DisposableObserver = Observer<any> & {
+  unsubscribe?: () => void;
+};
+
 export function createDeferredActor(
```

Comment:
this whole function is gross but it tries to keep the existing semantics as much as possible. The idea is to create an actor object as soon as possible - so we can give it to the user and don't change it afterward, we only call `.start()` on this from the outside and that call happens in the interpreter.

We still allow for non-lazy actors but we wrap them with a function `() => nonLazy` and "resolve" it here. So the existing code should mainly still work in the same but perhaps we should implement a warning for non-lazy actors so people could start refactoring their machines.